nlbq/api.py

```diff
@@ -10,19 +13,65 @@
 
 app = FastAPI()
 
+
+class DryRunResp(BaseModel):
+    statement: str
+    data: str
+    qpm: int
+
+
+class StatementReq(BaseModel):
+    statement: str
+    question: str
+
+
+class StatementResp(BaseModel):
+    text_results: str
+    html_results: str
+
+
+class AnswerReq(BaseModel):
+    question: str
+    statement: str
+    results: str
+
+
 @app.get("/api/dry_run")
-async def dry_run(q: Union[str, None] = None):
+async def dry_run(q: str) -> DryRunResp:
     """Process a text query and return the SQL statement, results, and explanation."""
     nlbq = NLBQ()
     statement = await nlbq.text_to_bq(q)
     data, qpm = nlbq.dry_run(statement)
-    return {
-        "response": {
-            "statement": statement,
-            "data": data,
-            "qpm": qpm,
-        }
-    }
+    return DryRunResp(statement=statement, data=data, qpm=qpm)
+
+
+@app.post("/api/run_statement")
+async def run_statement(request_data: StatementReq) -> StatementResp:
+    nlbq = NLBQ()
+    fields, rows = nlbq.execute(request_data.statement)
+    return StatementResp(
+        text_results=tabulate(rows, headers=fields),
+        html_results=tabulate(rows, headers=fields, tablefmt="html"),
+    )
+
+
+@app.post("/api/answer")
+async def answer(request_data: AnswerReq):
+    nlbq = NLBQ()
+    resp = await nlbq.answer(
+        request_data.question, request_data.statement, request_data.results
```

Comment:
was going to suggest passing just `request_data`, but it makes sense keeping the NLBQ methods more generic

Reply:
I'm not sure either. It'd be good to know the best practice here.